src/components/number-picker/number-picker.component.spec.ts

```diff
@@ -322,4 +324,23 @@ describe('Number Picker Component - value', () => {
         expect(numberPicker.classList.contains('ng-invalid')).toBe(false);
     });
 
+    it ('should display ux-number-picker-invalid class when value above max value and valid set to true', async() => {
+        component.valid = true;
+        component.value = 20;
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(input.value).toBe('20');
+        expect(numberPicker.classList.contains('ux-number-picker-invalid')).toBe(true);
+    });
+
+    it ('should not display ux-number-picker-invalid class when value above below value and valid set to false', async() => {
```

Comment:
I think there's a missing word here? Or maybe something like "when value is between min and max and..."?

Also could we have one for "should display ux-number-picker-invalid class when value is between min and max and valid set to true"?